### DISCUSSION TITLE
Best way to create users

### DISCUSSION THREAD

**Discussion by orestesgaolin** [hasura/graphql-engine#8723] [Question]
I found pretty cool [authentication tutorial](https://hasura.io/learn/graphql/hasura-authentication/integrations/firebase/) for Hasura and Firebase. It's relatively naive as the user accounts are being created by the client along with the sign up in Firebase Auth. 

I'd like to take this example a step further by delegating the user data creation in Hasura somewhere else. My first guess is to use callable Google Cloud Functions. This would let me leverage the Firebase Authentication and decouple the registration process from the client implementation. 

What do you think about this approach? How should I configure and authenticate my Cloud Function instance to perform mutations on the Hasura?

**Comment by adas98012** [comment]
Thanks for finding our tutorials useful. We have elaborate examples of using "serverless functions" in our docs, which also includes some boilerplates: https://hasura.io/docs/latest/event-triggers/serverless/#why-use-serverless-functions

Hope, that answers your question. We'll be happy to hear a successful outcome from your end.

**Reply by orestesgaolin** [reply]
Thanks a lot for the link! Wasn't aware of that before.

What we came up with was exposing REST endpoints from GraphQL query that we call via our cloud function. This way we can have some custom logic before creating the entry in the users table. That's really useful feature!
 
![CleanShot 2022-08-17 at 09 58 30@2x](https://user-images.githubusercontent.com/16854239/185066126-ef349896-fc01-4140-b8cb-731f913a801c.png)
